explorer/models/rootNode.ts

```diff
@@ -67,34 +67,24 @@ export class RootNode extends NodeBase {
                 let found: boolean = false;
 
                 const images: Docker.ImageDesc[] = await docker.getImageDescriptors(imageFilters);
+                images.sort((img1, img2) => {
+                    if (img1.Id > img2.Id) {
+                        return -1;
+                    } else if (img1.Id < img2.Id) {
+                        return 1;
+                    } else {
+                        return 0;
+                    }
+                });
 
                 if (!this._imageCache) {
                     this._imageCache = images;
+                    return;
                 }
 
-                if (this._imageCache.length !== images.length) {
-                    needToRefresh = true;
-                } else {
-                    // tslint:disable-next-line:prefer-for-of // Grandfathered in
-                    for (let i: number = 0; i < this._imageCache.length; i++) {
-                        let before: string = JSON.stringify(this._imageCache[i]);
-                        // tslint:disable-next-line:prefer-for-of // Grandfathered in
-                        for (let j: number = 0; j < images.length; j++) {
-                            let after: string = JSON.stringify(images[j]);
-                            if (before === after) {
-                                found = true;
-                                break;
-                            }
-                        }
-
-                        if (!found) {
-                            needToRefresh = true;
-                            break
-                        }
-                    }
-                }
-
-                if (needToRefresh) {
+                let imagesAsJson = JSON.stringify(images);
+                let cacheAsJson = JSON.stringify(this._imageCache);
+                if (imagesAsJson !== cacheAsJson) {
```

Comment:
Would you also need to sort `this._imageCache` ?

Reply:
It comes directly from images, which is being sorted here.  Changed it to _sortedImageCache to make that clear.